src/shared/tests.c

```diff
@@ -51,8 +51,12 @@ static void load_testdata_env(void) {
                 return;
         called = true;
 
-        assert_se(readlink_and_make_absolute("/proc/self/exe", &s) >= 0);
-        assert_se(path_extract_directory(s, &d) >= 0);
+        if (readlink_and_make_absolute("/proc/self/exe", &s) >= 0)
+                assert_se(path_extract_directory(s, &d) >= 0);
+        else {
+                log_debug_errno(errno, "Failed to determine executable's directory, using current directory.");
```

Comment:
readlink_and_make_absolute() is ours. Please use returned value `r = readlink_and_make_absolute(...)` as usual.